### PR TITLE
Configuring for heroku to use whitenoise

### PR DESCRIPTION
**Pull request by infobiac**

### CODE REVIEW

File: BuyBitcoin/settings.py [hunk 1]
```diff
@@ -92,7 +95,12 @@
 # Database
 # https://docs.djangoproject.com/en/1.9/ref/settings/#databases
 
-DATABASES = {}
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+    }
+}
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:104]: I don't think this doese anything

**Review by Neitsch — COMMENTED** [review]
Did you verify that this works with webpack?

**Comment by Neitsch** [comment]
lol
```
./BuyBitcoin/settings.py:152:1: E303 too many blank lines (3)
```

File: BuyBitcoin/settings.py (hunk 1 above)
Comment (infobiac) [reply BuyBitcoin/settings.py:104]: It does, switches local backend to sqlite, which means that you can easily create/destroy data. its the heroku recommended fashion of doing dbs (sqlite in dev, postgresql in prod).

**Comment by infobiac** [comment]
How can I verify if it works with webpack?

**Comment by Neitsch** [comment]
You can verify it by navigating to /home

The default is overwritten a couple lines below?

File: BuyBitcoin/settings.py [hunk 2]
```diff
@@ -104,8 +112,7 @@
         'PORT':     '',
     }
 else:
-    DATABASES['default'] = dj_database_url.config(
-        default="postgres://bitcoin:stupid_pw@localhost/buy_bitcoin")
+    DATABASES['default'].update(dj_database_url.config(conn_max_age=500))
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:115]: This line will overwrite line 100, no?

Comment (infobiac) [reply BuyBitcoin/settings.py:115]: Nope, this only selects if it finds the env

**Review by Neitsch — APPROVED** [review]
Verify that it works with Webpack, fix the lint, then we are good to go

File: BuyBitcoin/settings.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
-
+PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:19]: Just use BASE_DIR

File: BuyBitcoin/settings.py [hunk 4]
```diff
@@ -37,6 +37,8 @@
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
+    'whitenoise.runserver_nostatic',
+
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:41]: Remove  whitespace

File: BuyBitcoin/settings.py [hunk 5]
```diff
@@ -141,14 +148,11 @@
 USE_TZ = True
 
 
-# Static files (CSS, JavaScript, Images)
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:144]: Why did you remove the comment?

File: BuyBitcoin/settings.py [hunk 6]
```diff
@@ -141,14 +148,11 @@
...
-# https://docs.djangoproject.com/en/1.9/howto/static-files/
-
+STATIC_ROOT = os.path.join(PROJECT_ROOT, 'staticfiles')
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:151]: Change back to static

File: stocks/migrations/0001_initial.py [hunk 7]
```diff
@@ -17,22 +17,25 @@ class Migration(migrations.Migration):
         migrations.CreateModel(
             name='DailyStockQuote',
             fields=[
-                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('id', models.AutoField(auto_created=True,
+                                        primary_key=True, serialize=False, verbose_name='ID')),
```
Comment (Neitsch) [inline stocks/migrations/0001_initial.py:21]: Why did you reformat this?

**Review by Neitsch — COMMENTED** [review]
Plz address this when you fix the Heroku deployment
```
$ python manage.py collectstatic --noinput
       Traceback (most recent call last):
         File "manage.py", line 10, in <module>
           execute_from_command_line(sys.argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
           utility.execute()
         File "/app/.heroku/python/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
           self.fetch_command(subcommand).run_from_argv(self.argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
           self.execute(*args, **cmd_options)
         File "/app/.heroku/python/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
           output = self.handle(*args, **options)
         File "/app/.heroku/python/lib/python3.6/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 199, in handle
           collected = self.collect()
         File "/app/.heroku/python/lib/python3.6/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 115, in collect
           for path, storage in finder.list(self.ignore_patterns):
         File "/app/.heroku/python/lib/python3.6/site-packages/django/contrib/staticfiles/finders.py", line 112, in list
           for path in utils.get_files(storage, ignore_patterns):
         File "/app/.heroku/python/lib/python3.6/site-packages/django/contrib/staticfiles/utils.py", line 28, in get_files
           directories, files = storage.listdir(location)
         File "/app/.heroku/python/lib/python3.6/site-packages/django/core/files/storage.py", line 397, in listdir
           for entry in os.listdir(path):
       FileNotFoundError: [Errno 2] No such file or directory: '/tmp/build_12f46413e35b8c05732951ed3ebcc677/Neitsch-ASE4156-64880a1/BuyBitcoin/static'
 !     Error while running '$ python manage.py collectstatic --noinput'.
```

File: BuyBitcoin/settings.py [hunk 8]
```diff
@@ -159,6 +163,9 @@
 }
 
 
+# Simplified static file serving.
+# https://warehouse.python.org/project/whitenoise/
+STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
 # GraphQL
```
Comment (Neitsch) [inline BuyBitcoin/settings.py:169]: Add empty line in between

File: BuyBitcoin/settings.py (hunk 6 above)
Comment (infobiac) [reply BuyBitcoin/settings.py:151]: We have static and static files (two do different things - staticfiles go in the source folder, buybitcoin in this case, while static lives in root)

File: BuyBitcoin/settings.py (hunk 3 above)
Comment (infobiac) [reply BuyBitcoin/settings.py:19]: See comment on static vs staticfiles

File: BuyBitcoin/settings.py (hunk 6 above)
Comment (Neitsch) [reply BuyBitcoin/settings.py:151]: Then why did the Heroku deploy fail?
